Custom Resources Article

Manchick0 opened the pull request:
Introduced a comprehensive guide to custom resources, covering client and server-side data.

Commit:
Added the guide on custom resources for runtime loading.

Introduced a comprehensive guide to custom resources, covering client and server-side data. Implemented a `JsonDataLoader` example and updated sidebars, including their translations.

Commit:
Updated the class to have proper entrypoints

Commit:
Fixed tiny mistakes.

IMB11 commented on develop/custom-resources.md:
````diff
@@ -0,0 +1,126 @@
+and the `apply` method.
+
+```java
+public class FruitDataLoader extends JsonDataLoader {
````
Needs to be in the reference mod. Just because it uses mc classes which may change in the future (ResourceManager)

Manchick0 replied:
Sure! I added a new class, which is just empty. I really didn't want to do it though, that's why I decided to use raw code in the previous version.

IMB11 reviewed (changes requested):
Really good, just a minor nitpick.

Commit:
Added EmptyDataLoader class for documentation.

Introduced a new EmptyDataLoader class extending JsonDataLoader, intended solely for guidance in the 'custom resources' documentation.

Commit:
Fixed small syntax errors.

Commit:
Refactor data loader classes and update docs

Deleted EmptyDataLoader.java and introduced new Book and Fruit data loader classes for improved data handling. Updated documentation to reflect the changes, including new examples and clarifications on resource types and Codec usage.

Manchick0 commented:
Any updates?
